.github/workflows/pull-request.yaml

```diff
@@ -147,45 +150,79 @@ jobs:
   wasm:
     name: WASM
     runs-on: ubuntu-22.04
+    needs: generate
     steps:
       - name: Check out code
         uses: actions/checkout@v3
 
-      - name: Build and Test WASM
+      - id: changed-files
+        uses: tj-actions/changed-files@v28.0.0
+        with:
+          files: |
+            Makefile
+            wasm/
+            ast/
+            internal/compiler/
+            internal/planner/
+            internal/wasm/
+            test/wasm/
+            test/cases/
+
+      - name: Download generated artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: generated
+
+      - name: Build and Test Wasm
         run: make ci-wasm
         timeout-minutes: 15
+        if: steps.changed-files.outputs.any_changed == 'true'
+        env:
+          DOCKER_RUNNING: 0
+
+      - name: Build and Test Wasm SDK
+        run: make ci-go-wasm-sdk-e2e-test
+        timeout-minutes: 30
+        if: steps.changed-files.outputs.any_changed == 'true'
+        env:
+          DOCKER_RUNNING: 0
 
   check-generated:
     name: Check Generated
     runs-on: ubuntu-22.04
+    needs: generate
     steps:
       - name: Check out code
         uses: actions/checkout@v3
 
+      - name: Download generated artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: generated
+
       - name: Check Working Copy
         run: make ci-check-working-copy
         timeout-minutes: 15
-
-  wasm-go-sdk-e2e:
-    name: OPA Wasm SDK e2e
-    runs-on: ubuntu-22.04
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-
-      - name: Build and Test Wasm SDK
-        run: make ci-go-wasm-sdk-e2e-test
-        timeout-minutes: 30
+        env:
+          DOCKER_RUNNING: 0
```

Comment:
I *have* had the e2e tests catch mistakes in my PRs before (forgetting to propagate something over to the WASM build), but they do take a second to run. Moving this check to Nightly is probably fine. :+1:

Reply:
The wasm tests aren't actually moved to nightly; they are consolidated into one test. And that test is guarded: only run if some relevant files have changes.